include/ck/tensor_operation/gpu/grid/gridwise_batched_gemm_softmax_gemm_xdl_cshuffle_v1.hpp

```diff
@@ -410,6 +405,9 @@ struct GridwiseBatchedGemmSoftmaxGemm_Xdl_CShuffle
             return;
         }
 
+        // fetch origin N dim(before padding)
+        const index_t n_raw = b_grid_desc_bk0_n_bk1.GetTransforms()[I0].GetUpperLengths()[I0];
```

Comment:
This is VERY hacky and dangerious. Please don't do it.

Please make DeviceOp pass ```NRaw``` into GridOp as argument.

Reply:
Total agree. I'll do it.